### PR TITLE
Adding body property chartjs

### PR DESCRIPTION
**Pull request by mrjack88**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by lekoaf** [comment]
> Provide a URL to documentation or source code which provides context for the suggested changes

And fill in the rest of the template please.

File: types/chart.js/index.d.ts
Comment (adripanico) [inline types/chart.js/index.d.ts]: This is fine but could you define the interface of the body objects? According to [documentation](https://www.chartjs.org/docs/latest/configuration/tooltip.html#tooltip-model) it should be something like:

```
interface ChartTooltipBodyObject {
  before: string[];
  lines: string[];
  after: string[];
}
```

and then

```
interface ChartTooltipModel {
  bode: ChartTooltipBodyObject[];
  ...
}
```

**Review by adripanico — CHANGES_REQUESTED** [review]

File: types/chart.js/index.d.ts
Comment (lekoaf) [inline types/chart.js/index.d.ts]: You forgot to read the comment in the documentation.

```
// Body
// The body lines that need to be rendered
// Each object contains 3 parameters
// before: string[] // lines of text before the line with the color square
// lines: string[], // lines of text to render as the main item with color square
// after: string[], // lines of text to render after the main lines
```
https://www.chartjs.org/docs/latest/configuration/tooltip.html

Therefore, I suggest you create an interface like this:
(Better name might be needed)

```typescript
interface BodyProperties {
  before: string[]
  lines: string[]
  after: string[]
}
```

And then use that interface:

```typescript
body: BodyProperties[]
```

**Review by lekoaf — CHANGES_REQUESTED** [review]

**Comment by lekoaf** [comment]
Ahh, you beat me to it @adripanico. :)

**Comment by mrjack88** [comment]
Changes were taken from [Chart.js documentation page regarding tooltips](https://www.chartjs.org/docs/latest/configuration/tooltip.html#tooltip-model)

File: types/chart.js/index.d.ts
Comment (adripanico) [inline types/chart.js/index.d.ts]: missing semicolon

**Review by adripanico — CHANGES_REQUESTED** [review]

File: types/chart.js/index.d.ts
Comment (mrjack88) [reply types/chart.js/index.d.ts]: OMG I hate TS

**Review by lekoaf — APPROVED** [review]

**Comment by mrjack88** [comment]
Hey @adripanico , could you please take a look at new changes?
Thanks!

**Review by adripanico — APPROVED** [review]

**Comment by adripanico** [comment]
Thanks for you contribution!

**Comment by mrjack88** [comment]
Hi there guys, can someone please merge this PR?
@adripanico , @lekoaf ?

**Comment by lekoaf** [comment]
@mrjack88 no, neither of us have the ability to merge. We're "just" definition maintainers. Our job is to review the pull requests. It's the owners of DefinitelyTyped that does the merging. No idea why this isn't merged yet. They are usually quite quick.

**Comment by lekoaf** [comment]
The PR has a `Revisions needed` tag. That might be part of the problem.

**Review by lekoaf — APPROVED** [review]

**Comment by mrjack88** [comment]
@lekoaf any ideas why would it need revisions? I don't have a clue...

**Review by adripanico — APPROVED** [review]

**Comment by mrjack88** [comment]
Please, some maintainer help me understanding why this PR, which is 14 days old hasn't been merged yet? It still says need revisions... :( I really need this PR to be approved, so if there's something else I should do/check/test/change, can someone please let me know?
Thanks

**Comment by adripanico** [comment]
> Please, some maintainer help me understanding why this PR, which is 14 days old hasn't been merged yet? It still says need revisions... :( I really need this PR to be approved, so if there's something else I should do/check/test/change, can someone please let me know?
> Thanks

I really don't think any maintainer is going to read this message. Maybe it's easier to close this PR and create a new one with the same changes.

**Comment by mrjack88** [comment]
Thanks @adripanico I guess i'll do it like that. Long live to OSS :(

**Pull request closed without merging**
